source/docs/software/advanced-controls/introduction/approaches-to-ctrl-sys-design.rst

```diff
@@ -0,0 +1,70 @@
+Approaches to Designing Control Systems
+=======================================
+
+.. note:: This article includes sections of `Controls Engineering in FRC <https://file.tavsys.net/control/controls-engineering-in-frc.pdf>`__ by Tyler Veness with permission.
+
+When designing a control system for your robot, there are a number of different approaches to take. They range from very simple approaches, to advanced and complex ones. Each has *tradeoffs*. Some will work better than others in different situations, some require more mathematical analysis than others.
+
+Teams should prioritize picking the easiest strategy which enables success on the field. However, as you do experiments, keep in mind there is almost always a "next-step" to take to improve your field performance.
+
+Feedforward vs. Feedback
+------------------------
+
+There are two broad categories of strategies in control system design:
+
+**Feedforward** refers to the class of algorithms which incorporate knowledge of how the mechanism under control is *expected* to operate. Using this "model" of operation, the control input is chosen to make the mechanism get close to where it should be.
+
+**Feedback** refers to the class of algorithms which use sensors to *measure* what a mechanism is doing, and issue corrective commands to move a mechanism from where it actually is, to where you want it to be.
+
+They are not mutually exclusive - many techniques will use these concepts simultaneously.
+
+Start with Feedforward
+-----------------------
+
+Because most FRC mechanisms can be manufactured to be have well-defined behavior and can be easily modeled, it's generally encouraged to look into feedforward techniques first.
+
+How do you expect your system to behave?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first step in designing a control system is writing down information on how you expect your mechanism to behave.
+
+This step is done by combining one or more concepts you may be familiar with from physics: drawing free body diagrams of the forces that act on the mechanism, taking measurements of mass and moment of inertia from your CAD models, applying standard equations of how DC motors or pneumatic cylinders convert energy into mechanical force and motion, etc.
+
+The act of creating these mathematical descriptions is called *modeling* your system's behavior. The resulting set of equations are the *model* of how you expect the system to behave.
+
+Note that models do not have to be perfectly accurate to be useful. As we will see in later tuning exercises, even using a simple model of a mechanism can make the tuning effort much simpler.
```

Comment:
This implies the necessity of feedforward + feedback, which seems to contradict with above

Reply:
Even when using just feedback, modeling the system is a crucial step if you want to understand what approximate gains you will need and why.  Tuning without *some* sense for how the system behaves is effectively impossible.